src/status_im2/contexts/chat/lightbox/view.cljs

```diff
@@ -1,146 +1,91 @@
 (ns status-im2.contexts.chat.lightbox.view
   (:require
-    [quo2.core :as quo]
+    [clojure.string :as string]
     [quo2.foundations.colors :as colors]
     [react-native.core :as rn]
+    [react-native.orientation :as orientation]
     [react-native.platform :as platform]
     [react-native.reanimated :as reanimated]
+    [status-im2.contexts.chat.lightbox.common :as common]
     [utils.re-frame :as rf]
     [react-native.safe-area :as safe-area]
     [reagent.core :as reagent]
     [status-im2.contexts.chat.lightbox.style :as style]
-    [utils.datetime :as datetime]
     [status-im2.contexts.chat.lightbox.zoomable-image.view :as zoomable-image]
+    [status-im2.contexts.chat.lightbox.top-view :as top-view]
+    [status-im2.contexts.chat.lightbox.bottom-view :as bottom-view]
     [oops.core :refer [oget]]))
 
-(def flat-list-ref (atom nil))
-(def small-list-ref (atom nil))
-(def small-image-size 40)
-(def focused-image-size 56)
+(def seperator-width 16)
 
 (defn toggle-opacity
   [opacity-value border-value transparent? index]
   (let [opacity (reanimated/get-shared-value opacity-value)]
     (if (= opacity 1)
       (do
-        (reanimated/set-shared-value opacity-value (reanimated/with-timing 0))
+        (common/set-val-timing opacity-value 0)
         (js/setTimeout #(reset! transparent? (not @transparent?)) 400))
       (do
         (reset! transparent? (not @transparent?))
-        (js/setTimeout #(reanimated/set-shared-value opacity-value (reanimated/with-timing 1)) 50)
-        (js/setTimeout #(.scrollToIndex ^js @small-list-ref #js {:animated false :index index}) 100)))
-    (reanimated/set-shared-value border-value (reanimated/with-timing (if (= opacity 1) 0 12)))))
-
-(defn image
-  [message index _ {:keys [opacity-value border-value transparent?]}]
-  [rn/view {:style {:flex-direction :row}}
-   [zoomable-image/zoomable-image message index border-value
-    #(toggle-opacity opacity-value border-value transparent? index)]
-   [rn/view {:style {:width 16}}]])
+        (js/setTimeout #(common/set-val-timing opacity-value 1) 50)
```

Comment:
There are lot of delayed animations, these can also be handled while using derived value with the help of [withDelay](https://docs.swmansion.com/react-native-reanimated/docs/api/animations/withDelay)

Reply:
@Parveshdhull I have changed it to use `withDelay`. I do not think derived value is needed here?